src/index.ts

```diff
@@ -0,0 +1,125 @@
+
+const MONTHS_IN_YEAR = 12;
+const MINIMUM_PURCHASE_AMOUNT = 800;
+
+export type Result = {
+  year: number;
+  depreciationMonths: number;
+  depreciationAmount: number;
+  percentage: number;
+  startValue: number;
+  endValue: number;
+}
+
+const assertPurchaseAmount = (amount: number) => {
+  if (amount < 0) {
+    throw new TypeError('`purchaseAmount` is invalid.');
+  }
+};
+
+const assertPurchaseDate = (dateString: string) => {
+  if (isNaN(new Date(dateString).valueOf())) {
+    throw new TypeError('`purchaseDate` is invalid.');
+  }
+};
+
+const assertDepreciationYears = (years: number) => {
+  if (years < 1) {
+    throw new TypeError('`totalDepreciationYears` is invalid.');
+  }
+};
+
+const toFixedTwo = (num: number): number => Math.round((num + Number.EPSILON) * 100) / 100;
+
+const calculate = (
+  purchaseAmount: number,
+  totalDepreciationYears: number,
+  previousEndValue: number,
+  monthsLeft: number = MONTHS_IN_YEAR
+): {
+  depreciationAmount: number;
+  percentage: number;
+  startValue: number;
+  endValue: number;
+} => {
+  const depreciationAmount = toFixedTwo((purchaseAmount / totalDepreciationYears / MONTHS_IN_YEAR) * monthsLeft);
+  const percentage = toFixedTwo((depreciationAmount / purchaseAmount) * 100);
+  const newEndValue = toFixedTwo(previousEndValue - depreciationAmount);
+
+  return {
+    depreciationAmount,
+    percentage,
+    startValue: previousEndValue,
+    endValue: newEndValue,
+  };
+};
+
+const calculateDepreciation = (
+  purchaseAmount: number,
+  purchaseDate: string,
```

Comment:
🔧 I think it's better to use `Date` type for the date (instead of `string`) to move date parsing outside of the scope of that library

Reply:
thanks, it's updated https://github.com/kontist/asset-depreciation-calculator/pull/1/commits/b4bf8180a81198c65b6c635aa41e7ff4614a85f4